### DISCUSSION TITLE
How to use login.php as standalone page in PHP 8.2

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5026] [Q&A Core]
Because this is in login.php

`if(!defined('e_IFRAME')) define('e_IFRAME',true);`

login page doesn't have a header or footer and the user needs to use the back button to get to the homepage or any page. They often close the page because it can look like a modal window.

Years ago was this way recommended as fix:  add to theme.php

`if (e_PAGE == "login.php") define('e_IFRAME', false);`

Now it fails in PHP 8.2. because not defined e_PAGE.  

Is there another way like this?
`if ( defined('e_PAGE') &&  e_PAGE == "login.php") define('e_IFRAME', false);`

This should be theme developer stuff, not core one. 
 
At least all splash pages (login, signup, password forgotten) should have the same logic in this and the same way of templating (to cover both version of site - general and only members)  

Thanks

**Comment by roma-marshall** [comment]


**Reply by Jimmi08** [reply]
Thanks. The question was if there is another way how you recommended, but thanks for the effort.

**Reply by Deltik** [reply]
This answer is spam.

**Comment by Jimmi08** [comment]
> This answer is spam.

Yes, I know. And AI. But I liked the explanation. Very good for documentation :) Or learning English.
